source/extensions/filters/common/rbac/engine_impl.h

```diff
@@ -27,8 +27,15 @@ enum class EnforcementMode { Enforced, Shadow };
 
 class RoleBasedAccessControlEngineImpl : public RoleBasedAccessControlEngine, NonCopyable {
 public:
+  // TODO(Jojy): Adding `validation_visitor` as optional here. Reasons for this design choice:
+  //    - RBAC engine is not dependent on protobuf validator. The dependency is artificial.
+  //    - Adding a non-optional parameter would change the existing API and will have larger radius
+  //    of impact.
+  //    Will revisit it and consider alternatives like a separate API for setting the validator.
```

Comment:
I'm not sure I understand this comment. I would require validation visitor to be passed through in all places. Or if it doesn't make sense, just use a null validation visitor at callsites.